src/test/streams_tests.cpp

```diff
@@ -9,6 +9,18 @@
 
 BOOST_FIXTURE_TEST_SUITE(streams_tests, BasicTestingSetup)
 
+// This is a macro rather than a function so that failure
+// line numbers are reported correctly.
+#define BOOST_CHECK_THROW_STR(statement, exception, str)   \
```

Comment:
Rather not call our own macros `BOOST_*`...

Reply:
> Rather not call our own macros `BOOST_*` ...

Good point; I just removed this (new) macro because it's out of the scope for this PR, force-pushed (but didn't rebase).